src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -409,13 +414,37 @@ public void buttonClearDatePickerContestDateClicked() {
 
   /** Action when CVR file path button is clicked. */
   public void buttonCvrFilePathClicked() {
-    FileChooser fc = new FileChooser();
-    fc.setInitialDirectory(new File(FileUtils.getUserDirectory()));
-    fc.getExtensionFilters()
-        .add(new ExtensionFilter("Excel and JSON files", "*.xls", "*.xlsx", "*.json"));
-    fc.setTitle("Select cast vote record file");
+    File openFile = null;
+
+    switch (getChoiceElse(choiceCvrProvider, Provider.PROVIDER_UNKNOWN)) {
+      case "CDF": {
+        FileChooser fc = new FileChooser();
+        fc.setInitialDirectory(new File(FileUtils.getUserDirectory()));
+        fc.getExtensionFilters().add(new ExtensionFilter("JSON files", "*.json"));
+        fc.setTitle("Select CDF Cast Vote Record File");
+        openFile = fc.showOpenDialog(GuiContext.getInstance().getMainWindow());
+        break;
+      }
+      case "ES&S": {
+        FileChooser fc = new FileChooser();
+        fc.setInitialDirectory(new File(FileUtils.getUserDirectory()));
+        fc.getExtensionFilters()
+            .add(new ExtensionFilter("Excel files", "*.xls", "*.xlsx"));
+        fc.setTitle("Select ES&S Cast Vote Record File");
+        openFile = fc.showOpenDialog(GuiContext.getInstance().getMainWindow());
+        break;
+      }
+      case "Hart": {
+        DirectoryChooser dc = new DirectoryChooser();
+        dc.setInitialDirectory(new File(FileUtils.getUserDirectory()));
+        dc.setTitle("Select Hart Cast Vote Record Folder");
+        openFile = dc.showDialog(GuiContext.getInstance().getMainWindow());
+        break;
+      }
```

Comment:
I considered this, but unfortunately you can't use polymorphism for `FileChooser` and `DirectoryChooser` because they don't have a common class. I spose you could feed an enum or boolean in as a parameter to tell it to use one or the other though. You'd also need to parameterize both the `ExtensionFilter` and the title, so you're really not saving that many lines.

I say go ahead and go for it if you still think it makes sense in your Dominion PR once you've got it laid out there.